client/composables/settings/useSetting.ts

```diff
@@ -1,20 +1,20 @@
 import type { Ref } from 'vue'
+import useErrorToast from '~/composables/useErrorToast'
 
 export function useSetting<T>(
   setting: Ref<T>,
   mapSetting: (newSetting: T) => Partial<UserSettings>,
 ) {
   const { updateSettings } = useUserSettings()
-  const toast = useToast()
-  const { t, m } = useI18n()
+  const handleError = useErrorToast()
   const isError = ref<boolean>(false)
   const backupValue = shallowRef<T>(setting.value)
 
   watch(setting, (_, oldValue) => {
     backupValue.value = oldValue
   })
 
-  function patchSetting(newSetting: T) {
+  async function patchSetting(newSetting: T) {
```

Comment:
no need for async since you never use await

```suggestion
  function patchSetting(newSetting: T) {
```